skaha/src/main/java/org/opencadc/skaha/session/GetAction.java

```diff
@@ -176,86 +179,125 @@ private ResourceStats getResourceStats() throws Exception {
+                            if (!hasName && !hasCapacity && !hasCores) {
+                                nodeName = parts[1];
+                                hasName = true;
+                            } else {
+                                throw new IllegalStateException("Unexpected 'Name:' line");
+                            }
+                        } else if ("Capacity:".equals(parts[0])) {
+                            if (hasName && !hasCapacity && !hasCores) {
+                                hasCapacity = true;
+                            } else {
+                                throw new IllegalStateException("Unexpected 'Capabity:' line");
+                            }
+                        } else if (hasCapacity) {
+                            if ("cpu:".equals(parts[0])) {
+                                if (hasName && !hasCores) {
+                                    // number of cores from "Capabity.cpu"
+                                    int cores = Integer.parseInt(parts[1]);
+                                    resources[0] = cores;
+                                    hasCores = true;
+                                } else {
+                                    throw new IllegalStateException("Unexpected 'cpu:' line");
+                                }
+                            } else {
+                                // "memory:" equals parts[0] 
+                                if (hasName && hasCores) {
+                                    // finish processing the resources of a node
+                                    // amount of RAM from "Capabity.memory" is in Ki
+                                    int ram = Integer.parseInt(parts[1].replaceAll("[^0-9]", "").trim());
+                                    resources[1] = ram;
+                                    nodeToResourcesMap.put(nodeName, resources);
+                                    nodeName = "";
+                                    hasName = false;
+                                    hasCapacity = false;
+                                    hasCores = false;
+                                    resources = new int[2];
+                                } else {
+                                    throw new IllegalStateException("Unexpected 'memory:' line");
+                                }
+                            }
+                        }
+                    }
                 }
 
-                if (!hasRAM && line.indexOf("memory:") >= 0) {
-                    String[] parts = line.split(":");
-                    // amount of RAM from "Capabity.memory" in Ki
-                    int ram = Integer.parseInt(parts[1].replaceAll("[^0-9]", "").trim())/1048576;
-                    log.debug("Available memory in node " + nodeName + ": " + ram + " Gi");
-                    resources[1] = ram;
-                    hasRAM = true;
+                if (hasName || hasCapacity || hasCores) {
```

Comment:
Perhaps the logic is right, but it doesn't read very well.  You'd think this if statement means that you have at least one of name, capacity, or cores.  But then an exception is thrown saying you don't have those, so is confusing.

Reply:
I am going to add a comment to explain this check. Basically at this point, we have processed all nodes. All flags should have been cleared. If not, then there is a problem. Either the output from kubectl is incorrect (e.g. missing some lines) or there is a problem in our logic.

Reply:
Code removed after the latest update.